easycv/apis/export.py

```diff
@@ -372,3 +448,199 @@ def replace_syncbn(backbone_cfg):
             backbone_cfg['norm_cfg']['type'] = 'IBN'
 
     return backbone_cfg
+
+if LooseVersion(torch.__version__) >= LooseVersion('1.7.0'):
+
+    @torch.jit.script
+    class PreProcess:
+        """Process the data input to model.
+
+        Args:
+            target_size (Tuple[int, int]): output spatial size.
+            keep_ratio (bool): Whether to keep the aspect ratio when resizing the image.
+        """
+
+        def __init__(self,
+                     target_size: Tuple[int, int],
+                     keep_ratio: bool = True):
+            self.target_size = target_size
+            self.keep_ratio = keep_ratio
+
+        def __call__(
+            self, image: torch.Tensor
+        ) -> Tuple[torch.Tensor, Dict[str, Tuple[float, float]]]:
+            """
+            Args:
+                image (torch.Tensor): image format should be [H, W, C]
+            """
+            input_h, input_w = self.target_size
+            image = image.permute(2, 0, 1)
+            image = torch.unsqueeze(image, 0)
+            ori_h, ori_w = image.shape[-2:]
+
+            if not self.keep_ratio:
+                out_image = t_f.resize(image, [input_h, input_w])
+                pad_l, pad_t, scale = 0, 0, 1.0
+            else:
+                scale = min(input_h / ori_h, input_w / ori_w, 1.0)
+                resize_h, resize_w = int(ori_h * scale), int(ori_w * scale)
+                pad_h, pad_w = input_h - resize_h, input_w - resize_w
+                pad_l, pad_t = pad_w // 2, pad_h // 2
+                pad_r, pad_b = pad_w - pad_l, pad_h - pad_t
+                out_image = t_f.resize(image, [resize_h, resize_w])
+                out_image = t_f.pad(
+                    out_image, [pad_l, pad_t, pad_r, pad_b], fill=114)
+
+            h, w = out_image.shape[-2:]
+            output_info = {
+                'pad': (float(pad_l), float(pad_t)),
+                'scale_factor': (float(scale), float(scale)),
+                'ori_img_shape': (float(ori_h), float(ori_w)),
+                'img_shape': (float(h), float(w))
+            }
+            return out_image, output_info
+
+    @torch.jit.script
+    class DetPostProcess:
+        """Process output values of detection models.
```

Comment:
Hi, Is the output here done after the `nms`?

Reply:
yes, nms has been done in the forward function.

Reply:
yes, nms has been done in the forward function.